File: mac-app/Native/Vault/Models/Document.swift
import Foundation

struct Document {
    let id: String
    var title: String
    var body: String
    let createdAt: String
    var updatedAt: String

    init(id: String = UUID().uuidString,
         title: String = "",
         body: String = "",
         createdAt: String = ISO8601DateFormatter().string(from: Date()),
         updatedAt: String = ISO8601DateFormatter().string(from: Date())) {
        self.id = id
        self.title = title
        self.body = body
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

File: mac-app/Web/src/types/document.ts
export interface DocumentSummary {
  id: string;
  title: string;
  body_snippet: string;
  createdAt: string;
  updatedAt: string;
  tags: import('./tag').Tag[];
  categories: import('./category').Category[];
}

export interface Document extends DocumentSummary {
  body: string;
  assets: import('./asset').AssetSummary[];
}

export interface CreateDocumentPayload {
  title?: string;
  body?: string;
}

export interface UpdateDocumentPayload {
  title?: string;
  body?: string;
}
